trie/sync.go

```diff
@@ -208,9 +233,26 @@ func (s *Sync) Missing(max int) []common.Hash {
 		// Item is allowed to be scheduled, add it to the task list
 		s.queue.Pop()
 		s.fetches[depth]++
-		requests = append(requests, item.(common.Hash))
+
+		hash := item.(common.Hash)
+		if req, ok := s.nodeReqs[hash]; ok {
+			nodeHashes = append(nodeHashes, hash)
+
+			// If the hash is from the account trie, append a single item, if it
+			// is from the a storage trie, append a tuple. Note, the length 64 is
+			// clashing between account leaf and storage root. It's fine thpough
+			// because having a trie node at 64 depth means a hash collition was
+			// found and we're long dead.
+			if len(req.path) < 64 {
+				nodePaths = append(nodePaths, SyncPath{hexToCompact(req.path)})
+			} else {
+				nodePaths = append(nodePaths, SyncPath{hexToKeybytes(req.path[:64]), hexToCompact(req.path[64:])})
+			}
```

Comment:
Now that you have a proper type, you could move this logic in there: 
```suggestion
                        nodePaths = append(nodePaths, NewSyncPath(req.path))
```

Reply:
Done